mono/mini/mini.h

```diff
@@ -3016,4 +3016,69 @@ mono_arch_load_function (MonoJitICallId jit_icall_id);
 MonoGenericContext
 mono_get_generic_context_from_stack_frame (MonoJitInfo *ji, gpointer generic_info);
 
+#if defined(__APPLE__) && defined(__arm64__)
```

Comment:
Won't this enable all this for existing iOS devices too? Looks like `__builtin_available` guards against much actually happening, but this will still add some superfluous code or maybe not compile on non-Silicon iOS